sdk/identity/Azure.Identity/src/AzureArcManagedIdentitySource.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Identity
+{
+    internal class AzureArcManagedIdentitySource : ManagedIdentitySource
+    {
+
+        private const string IdentityEndpointInvalidUriError = "The environment variable IDENTITY_ENDPOINT contains an invalid Uri.";
+        private const string NoChallengeErrorMessage = "Did not receive expected WWW-Authenticate header in the response from Azure Arc Managed Identity Endpoint.";
+        private const string InvalidChallangeErrorMessage = "The WWW-Authenticate header in the response from Azure Arc Managed Identity Endpoint did not match the expected format.";
+        private const string ArchApiVersion = "2019-11-01";
+
+        private readonly string _clientId;
+        private readonly Uri _endpoint;
+
+        public static ManagedIdentitySource TryCreate(CredentialPipeline pipeline, string clientId)
+        {
+            string identityEndpoint = EnvironmentVariables.IdentityEndpoint;
+            string imdsEndpoint = EnvironmentVariables.ImdsEndpoint;
+
+            // if BOTH the env vars IDENTITY_ENDPOINT and IMDS_ENDPOINT are set the MsiType is Azure Arc
+            if (string.IsNullOrEmpty(identityEndpoint) || string.IsNullOrEmpty(imdsEndpoint))
+            {
+                return default;
+            }
+
+            Uri endpointUri;
+            try
+            {
+                endpointUri = new Uri(identityEndpoint);
+            }
+            catch (FormatException ex)
+            {
+                throw new AuthenticationFailedException(IdentityEndpointInvalidUriError, ex);
+            }
+
+            return new AzureArcManagedIdentitySource(pipeline, endpointUri, clientId);
+        }
+
+        private AzureArcManagedIdentitySource(CredentialPipeline pipeline, Uri endpoint, string clientId) : base(pipeline)
+        {
+            _endpoint = endpoint;
+            _clientId = clientId;
+        }
+
+        protected override Request CreateRequest(string[] scopes)
+        {
+            // covert the scopes to a resource string
+            string resource = ScopeUtilities.ScopesToResource(scopes);
```

Comment:
I recall hearing the endpoint accepts v2 scopes; can someone with a test environment confirm (@mccoyp @g2vinay @catalinaperalta)?

Reply:
I can confirm that v2 scopes work in my environment

Reply:
If by V2 scope you mean, sending the scope with /.default, ex: "https://valut.azure.net/.default"
Then the initial request works fine to get the secret key, but the auth req with the Authorization header set, comes back with 400 when scope with /.default is sent in my Arc Vm.

Reply:
I was able to get a deleted secret with something like:
```Python
credential = ManagedIdentityCredential()
token = credential.get_token("https://vault.azure.net/.default")
client = SecretClient("vault_url", credential)
client.get_deleted_secret("TestSecret")
```
Does this work to validate the behavior, or is there another way I should test v2 scopes?

Reply:
The public API for `ManagedIdentityCredential` does take in scopes, but these are internally converted to resource strings. Testing it out it I confirmed the endpoint only accepts resource strings similarly to all other MI endpoints, @g2vinay  also independently confirmed this.